mk/cargo.sh

```diff
@@ -108,6 +109,13 @@ case $target in
       export CARGO_TARGET_X86_64_UNKNOWN_LINUX_MUSL_RUSTFLAGS="$rustflags_self_contained"
     fi
     ;;
+  s390x-unknown-linux-gnu)
+    export CC_s390x_unknown_linux_gnu=clang-$llvm_version
+    export AR_s390x_unknown_linux_gnu=llvm-ar-$llvm_version
+    export CFLAGS_s390x_unknown_linux_gnu="--sysroot=/usr/s390x-linux-gnu"
+    export CARGO_TARGET_S390X_UNKNOWN_LINUX_GNU_LINKER=s390x-linux-gnu-gcc
+    export CARGO_TARGET_S390X_UNKNOWN_LINUX_GNU_RUNNER="$qemu_s390x"
+    ;;
```

Comment:
Does clang work? Do you have a preferred toolchain, GCC vs clang? Generally I prefer clang because it is much lighter weight. If your team has an official preference for GCC then we can keep using GCC.

Reply:
I mean, is there a way to avoid using `s390x-linux-gnu-gcc` for linking and instead use rust-lld or clang as the linker?

Reply:
lld doesn't support the s390x target at this point.  It should be possible to use the clang driver to invoke the (system) linker, but that'll just do the same as the gcc driver.  Do you want me to make that change anyway?

Reply:
> lld doesn't support the s390x target at this point. It should be possible to use the clang driver to invoke the (system) linker, but that'll just do the same as the gcc driver. Do you want me to make that change anyway?

No, it's OK this way. Thanks!